src/quickstart.ts

```diff
@@ -0,0 +1,111 @@
+import * as fsNode from 'fs';
+import * as path from 'path';
+
+import {fs} from './pr';
+
+export enum ProjectType {
+  Library,
+  Exectable
+}
+
+export interface ProjectTypeDesciptor {
+  label: string;
+  type: ProjectType;
+  description: string;
+}
+
+export const projectTypeDescriptions: ProjectTypeDesciptor[] = [
+  {label: 'Library', type: ProjectType.Library, description: 'Create a library'},
+  {label: 'Executable', type: ProjectType.Exectable, description: 'Create an executable'}
+];
+
+export class CMakeQuickStart {
```

Comment:
I tried to redesign it as a function collection but in that case each function will get a quickstart parameter object. This is like writing C function with structurs to get a object oriented design. I would leave it as a class.

Reply:
Not really. Object orientation is more than just composing attributes together. Having a parameter object is more of a "Named-Parameter Idiom" for JS/TS. There are no invariants on the fields of the parameter collection (other than their names and types), neither are there methods acting upon it. Usage might look like this:

```typescript
const qs_res = await qs.quickStart({
  sourceDir: "<some-path>",
  type: ProjectType.Library,
  name: "MyProject",
});
```

And the usage with the UI might look like this:

```typescript
// Show the UI to get the parameters for a quickstart
const qs_params = await qs.getQuickStartOptionsViaUI({sourceDir: this.sourceDir});
if (!qs_params) {
  // Use cancelled
  return;
}
const qs_res = await qs.quickStart(qs_params);
await vscode.window.showTextDocument(qs_res.mainListFile);
await this.configure();
```

Reply:
I changed the quickstart class by quickstart.createProject.

Do we want to separate UI from UI workflow? UI can only be tested by vscode integration test. The workflow chould be tested without vscode intergation. If we use always the same UI elements we could check our expectations against vscode by small tests for the UI elements. This would allow to reduce the needed extension tests.